bin/specextract

```diff
@@ -1823,1511 +1982,123 @@ def specextract(args):
-
-    if "NONE" == bggtype:
-        bgdogroup = False
-
-    else:
-        bgdogroup = True
-
-        if "BIN" == bggtype:
-            bgbinspec = bggspec
-            bggval = ""
-
-        else:
-            bgbinspec = ""
-            bggval = bggspec
-
-
-    #--------------------------------------------------------------------
-    # Determine whether or not to combine source output spectra, and any
-    # associated background spectra and response files, if user has input
-    # a stack of source event files.
+    # co-add datasets
     #
-    # docombine         # flag set if combining
-    #----------------------------------------------------------
+    ###########################
 
     if combine == "yes":
-        if src_count < 2:
-            print("Warning: There are fewer than two source event files specified in the 'infile' parameter; the 'combine=yes' setting will be ignored.\n")
+        combine_outroot = specdicts["src1"]["full_outroot"].rstrip("_src1")
+        dobg = specdicts["src1"]["dobg"]
+        dobkgresp = specdicts["src1"]["dobkgresp"]
 
-            docombine = False
+        sdvals = specdicts.values()
 
-            cs_src_spectra = [0]
-            cs_src_arfs = [0]
-            cs_src_rmfs = [0]
-            cs_bkg_spectra = [0]
-            cs_bkg_arfs = [0]
-            cs_bkg_rmfs = [0]
+        src_spec = [d["phafile"] for d in sdvals if d["srcbkg"]=="src"]
+        src_arf = [d["ARF"] for d in sdvals if d["srcbkg"] == "src"]
+        src_rmf = [d["RMF"] for d in sdvals if d["srcbkg"] == "src"]
+        
+        if dobg:
+            bkg_spec = [d["phafile"] for d in sdvals if d["srcbkg"]=="bkg"]
 
+            if dobkgresp:
+                bkg_arf = [d["ARF"] for d in sdvals if d["srcbkg"]=="bkg"]
+                bkg_rmf = [d["RMF"] for d in sdvals if d["srcbkg"]=="bkg"]
+            else:
+                bkg_arf = None
+                bkg_rmf = None
         else:
-            docombine = True
-
-            cs_src_spectra = list(range(src_count))
-            cs_src_arfs = list(range(src_count))
-            cs_src_rmfs = list(range(src_count))
-
-            sphafiles = list(range(src_count))
-            sarffiles = list(range(src_count))
-            srmffiles = list(range(src_count))
+            bkg_spec = None
+            bkg_arf = None
+            bkg_rmf = None    
+        
+        coadd(combine_outroot,
+              src_spec,bkg_spec,
+              src_arf,bkg_arf,
+              src_rmf,bkg_rmf,
+              dobg,dobkgresp,
+              verbose,clobber)
 
-            if dobg:
-                cs_bkg_spectra = list(range(src_count))
-                cs_bkg_arfs = list(range(src_count))
-                cs_bkg_rmfs = list(range(src_count))
 
-                bphafiles = list(range(src_count))
-                barffiles = list(range(src_count))
-                brmffiles = list(range(src_count))
+            
+if __name__  == "__main__":
+    try:
```

Comment:
If you add

```
import ciao_contrib.logger_wrapper as lw

...

@lw.handle_ciao_errors(toolname, __revision__)
def run_specextract(..):
...
```

then you don't need this `try/except` call.

Reply:
And I see `run_specextract` has this, so you don't need it.

Reply:
👍